src/test/ui/type-alias-impl-trait/structural-match-no-leak.rs

```diff
@@ -12,7 +12,8 @@ const LEAK_FREE: Bar = leak_free();
 fn leak_free_test() {
     match LEAK_FREE {
         LEAK_FREE => (),
-        //~^ `Bar` cannot be used in patterns
+        //~^ ERROR constant pattern depends on a generic parameter
```

Comment:
this error isn't ideal but could be reverted to the previous error by splitting the `ErrorHandled::TooGeneric` into two variants, one when encountering a generic parameter and one when encountering an opaque type with `Reveal::UserFacing`.